components/openvsx-proxy/pkg/modifyresponse.go

```diff
@@ -93,8 +129,26 @@ func (o *OpenVSXProxy) ModifyResponse(r *http.Response) error {
 	}
 
 	// no error (status code < 500)
+	body := rawBody
 	contentType := r.Header.Get("Content-Type")
 	if strings.HasPrefix(contentType, "application/json") {
+		if strings.EqualFold(r.Header.Get("Content-Encoding"), "gzip") {
+			gzipReader, err := gzip.NewReader(ioutil.NopCloser(bytes.NewBuffer(rawBody)))
+			if err != nil {
+				log.WithFields(logFields).WithError(err)
+				return nil
+			}
+
+			body, err = ioutil.ReadAll(gzipReader)
+			if err != nil {
+				log.WithFields(logFields).WithError(err).Error("error reading response body")
+				return nil
+			}
+			gzipReader.Close()
+
+			r.Header.Del("Content-Encoding")
```

Comment:
Should not we compress the final result then to reduce load on clients?

Reply:
I wans't sure about it, but now that you mention it I think it makes sense